packages/wdio-browserstack-service/src/util.ts

```diff
@@ -97,7 +106,157 @@ export function getParentSuiteName(fullTitle: string, testSuiteTitle: string): s
     return parentSuiteName.trim()
 }
 
-export async function launchTestSession (options: BrowserstackConfig & Options.Testrunner, config: Options.Testrunner, bsConfig: UserConfig) {
+function filterPII(userConfig: Options.Testrunner) {
+    const configWithoutPII = JSON.parse(JSON.stringify(userConfig));
+    ['user', 'username', 'key', 'accessKey'].forEach(key => delete configWithoutPII[key])
+    const finalServices = []
+    try {
+        for (const serviceArray of configWithoutPII.services) {
+            if (Array.isArray(serviceArray) && serviceArray.length >= 2 && serviceArray[0] === 'browserstack') {
+                for (let idx=1; idx<serviceArray.length; idx++) {
+                    ['user', 'username', 'key', 'accessKey'].forEach(key => delete serviceArray[idx][key])
+                }
+                finalServices.push(serviceArray)
+                break
+            }
+        }
+    } catch (err) {
+        /* Wrong configuration like strings instead of json objects could break this method, needs no action */
+    }
+    configWithoutPII.services = finalServices
+    return configWithoutPII
+}
+
+function setCredentialsForCrashReportUpload(options: BrowserstackConfig & Options.Testrunner, config: Options.Testrunner) {
+    credentialsForCrashReportUpload = {
+        username: getObservabilityUser(options, config),
+        password: getObservabilityKey(options, config)
+    }
+    process.env.CREDENTIALS_FOR_CRASH_REPORTING = JSON.stringify(credentialsForCrashReportUpload)
+}
+
+export function setConfigDetails(userConfig: Options.Testrunner, capabilities: Capabilities.RemoteCapability, options: BrowserstackConfig & Options.Testrunner) {
+    const configWithoutPII = filterPII(userConfig)
+    userConfigForReporting = {
+        framework: userConfig.framework,
+        services: configWithoutPII.services,
+        capabilities: capabilities,
+        env: {
+            'BROWSERSTACK_BUILD': process.env.BROWSERSTACK_BUILD,
+            'BROWSERSTACK_BUILD_NAME': process.env.BROWSERSTACK_BUILD_NAME,
+            'BUILD_TAG': process.env.BUILD_TAG
+        }
+    }
+    process.env.USER_CONFIG_FOR_REPORTING = JSON.stringify(userConfigForReporting)
+    setCredentialsForCrashReportUpload(options, userConfig)
+}
+
+function processError(error: any, fn: Function, args: any) {
```

Comment:
can we type this?

Reply:
- `error`: The first argument is the `error` that gets passed from a catch block. Have checked the type we receive in catch block, that was of unknown type, so had to keep any as it is not a deterministic type
- `args`: These are the args that received to a function, and as this is a generic function and type of args not known before hand so had to keep any. Have moved this to `any[]` type, that will be more specific